cli/tests/program.rs

```diff
@@ -2009,3 +2054,180 @@ fn create_buffer_with_offline_authority<'a>(
+        is_final: false,
+        max_len: Some(max_len),
+        skip_fee_check: false,
+        compute_unit_price,
+    });
+    config.output_format = OutputFormat::JsonCompact;
+    let response = process_command(&config);
+    let json: Value = serde_json::from_str(&response.unwrap()).unwrap();
+    let program_pubkey_str = json
+        .as_object()
+        .unwrap()
+        .get("programId")
+        .unwrap()
+        .as_str()
+        .unwrap();
+    assert_eq!(
+        program_keypair.pubkey(),
+        Pubkey::from_str(program_pubkey_str).unwrap()
+    );
+    let program_account = rpc_client.get_account(&program_keypair.pubkey()).unwrap();
+    assert_eq!(program_account.lamports, minimum_balance_for_program);
+    assert_eq!(program_account.owner, bpf_loader_upgradeable::id());
+    assert!(program_account.executable);
+    let signature_statuses = rpc_client
+        .get_signatures_for_address_with_config(
+            &keypair.pubkey(),
+            GetConfirmedSignaturesForAddress2Config {
+                commitment: Some(CommitmentConfig::confirmed()),
+                ..GetConfirmedSignaturesForAddress2Config::default()
+            },
+        )
+        .unwrap();
+    let signatures: Vec<_> = signature_statuses
+        .into_iter()
+        .rev()
+        .map(|status| Signature::from_str(&status.signature).unwrap())
+        .collect();
+
+    fn fetch_and_decode_transaction(rpc_client: &RpcClient, signature: &Signature) -> Transaction {
+        rpc_client
+            .get_transaction_with_config(
+                signature,
+                RpcTransactionConfig {
+                    encoding: Some(UiTransactionEncoding::Base64),
+                    commitment: Some(CommitmentConfig::confirmed()),
+                    ..RpcTransactionConfig::default()
+                },
+            )
+            .unwrap()
+            .transaction
+            .transaction
+            .decode()
+            .unwrap()
+            .into_legacy_transaction()
+            .unwrap()
+    }
+
+    assert!(signatures.len() >= 4);
+    let initial_tx = fetch_and_decode_transaction(&rpc_client, &signatures[1]);
+    let write_tx = fetch_and_decode_transaction(&rpc_client, &signatures[2]);
+    let final_tx = fetch_and_decode_transaction(&rpc_client, signatures.last().unwrap());
+
+    if let Some(compute_unit_price) = compute_unit_price {
+        for tx in [&initial_tx, &write_tx, &final_tx] {
+            for i in [0, 1] {
+                assert_eq!(
+                    tx.message.instructions[i].program_id(&tx.message.account_keys),
+                    &compute_budget::id()
+                );
+            }
+
+            match try_from_slice_unchecked(&tx.message.instructions[0].data) {
+                Ok(ComputeBudgetInstruction::SetComputeUnitPrice(price))
+                    if price == compute_unit_price => {}
+                ix => assert!(false, "unexpected ix {ix:?}"),
+            }
```

Comment:
nit: I believe all of these checks could use [`matches!`](https://doc.rust-lang.org/std/macro.matches.html) to be simpler, ie:

```suggestion
           assert!(
               matches!(
                   try_from_slice_unchecked(&tx.message.instructions[0].data),
                   Ok(ComputeBudgetInstruction::SetComputeUnitPrice(price)) if price == compute_unit_price
               ),
               "unexpected ix {ix:?}"
           );
```